mlrun/api/crud/marketplace.py

```diff
@@ -183,25 +194,55 @@ def get_source_catalog(
     def get_item(
         self,
         source: MarketplaceSource,
-        item_name,
-        channel,
-        version=None,
-        tag=None,
-        force_refresh=False,
+        item_name: str,
+        version: Optional[str] = None,
+        tag: Optional[str] = None,
+        force_refresh: bool = False,
     ) -> MarketplaceItem:
-        catalog = self.get_source_catalog(source, channel, version, tag, force_refresh)
-        items = [item for item in catalog.catalog if item.metadata.name == item_name]
-        if not items:
+        """
+        Retrieve item from source. The item is filtered by tag and version.
+
+        :param source:          Marketplace source object
+        :param item_name:       name of the item to retrieve
+        :param version:         version of the item
+        :param tag:             tag of the item
+        :param force_refresh:   if True, the catalog will be loaded from source always,
+                                otherwise will be pulled from db (if loaded before)
+
+        :return: marketplace item object
+
+        :raise if the number of collected items from catalog is not exactly one.
+        """
+        catalog = self.get_source_catalog(source, version, tag, force_refresh)
+        items = self._get_catalog_items_filtered_by_name(catalog.catalog, item_name)
+        num_items = len(items)
+
+        if not num_items:
             raise mlrun.errors.MLRunNotFoundError(
-                f"Item not found. source={item_name}, channel={channel}, version={version}"
+                f"Item not found. source={item_name}, version={version}"
             )
-        if len(items) > 1:
+        if num_items > 1:
             raise mlrun.errors.MLRunInvalidArgumentError(
                 "Query resulted in more than 1 catalog items. "
```

Comment:
I understand the validation, I dont understand the scenario. that is for when you query and catch mutliple items? if so, why would it happen? any idea?

Reply:
I guess wrong use of marketplace conventions, since the user can create its own marketplace and change it as he pleased